### PR TITLE
Clean up storageService.CreateContainerStorage

### LINKED ISSUE #20411: Surprising code in storageService.CreateContainerStorage
### Issue Description

I found some locally-surprising code, looking at the https://github.com/containers/podman/blob/0b0f128bfb682e871f40c80dcc63275b1599ae4f/libpod/storage.go#L70 condition:
- If `imageName` is set… it isn’t used for anything
- Actually, just the opposite, https://github.com/containers/podman/blob/0b0f128bfb682e871f40c80dcc63275b1599ae4f/libpod/storage.go#L98-L100 is dead on that path, and https://github.com/containers/podman/blob/0b0f128bfb682e871f40c80dcc63275b1599ae4f/libpod/storage.go#L101 seems to do nothing (if `imageID` really carries an image ID)
- `imageConfig` is _only_ set if `imageName` is non-empty, I wouldn’t expect the two to be related.

I didn’t check the callers / origins of the data. It’s quite possible that this does not matter in practice because `imageName`/`imageID` are always set in some particular way.

### Steps to reproduce the issue

N/A

### Describe the results you received

N/A

### Describe the results you expected

N/A

### podman info output

```yaml
N/A
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mtrmac**
Fixes #20411. See individual commits for details.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: libpod/storage.go [hunk 1]
```diff
@@ -69,39 +69,25 @@ func (metadata *RuntimeContainerMetadata) SetMountLabel(mountLabel string) {
 // CreateContainerStorage creates the storage end of things.  We already have the container spec created
 // TO-DO We should be passing in an Image object in the future.
 func (r *storageService) CreateContainerStorage(ctx context.Context, systemContext *types.SystemContext, imageName, imageID, containerName, containerID string, options storage.ContainerOptions) (_ ContainerInfo, retErr error) {
-	var imageConfig *v1.Image
-	if imageName != "" {
```
Comment (Luap99) [inline libpod/storage.go:73]: removing this breaks `--rootfs`, no? This should still check for `imageID != ""` which is the case if there is no image.

Comment (mtrmac) [reply libpod/storage.go:73]: Oops. Thanks, fixed.

**Comment by rhatdan** [comment]
LGTM

**Review by vrothberg — APPROVED** [review]
LGTM
/hold cancel

**Pull request merged**